app/libs/js/LEAF/formGrid.js

```diff
@@ -61,6 +60,107 @@ var LeafFormGrid = function (containerID, options) {
     showIndex = false;
   }
 
+  /**
+   * @param values (required) object of cells and names to generate grid
+   * @param showScriptTags (default false) whether to display script tags
+   * @memberOf LeafFormGrid
+   * Returns copy of values with cells property html entities decoded
+   */
+  function decodeCellHTMLEntities(values, showScriptTags = false) {
+    let gridInfo = { ...values };
+    if (gridInfo?.cells) {
+      let cells = gridInfo.cells.slice();
+      cells.forEach((arrRowVals, ci) => {
+        arrRowVals = arrRowVals.map((v) => {
+          v = v.replaceAll("<", "&lt;"); //handle old data values
+          v = v.replaceAll(">", "&gt;");
+          let elDiv = document.createElement("div");
+          elDiv.innerHTML = v;
+          let text = elDiv.innerText;
+          if (showScriptTags !== true)
+            text = text.replaceAll(
+              /(<script[\s\S]*?>)|(<\/script[\s\S]*?>)/gi,
+              ""
+            );
+          return text;
+        });
+        cells[ci] = arrRowVals.slice();
+      });
+      gridInfo.cells = cells;
+    }
+    return gridInfo;
+  }
+
+  /**
+   * @param values (required) object of cells and names to generate grid
+   * @memberOf LeafFormGrid
+   */
+  function printTableReportBuilder(values, columnValues) {
+    // remove unused columns
+    values = decodeCellHTMLEntities(values);
+    if (columnValues !== null && columnValues !== undefined) {
+      values.format = values.format.filter(function (value) {
+        return columnValues.includes(value.id);
+      });
+    }
+
+    var gridBodyBuffer = "";
+    var gridHeadBuffer = "";
+    var rows = values.cells === undefined ? 0 : values.cells.length;
+    var columns = values.format.length;
+    var columnOrder = [];
+    var delim = '<span class="nodisplay">^;</span>'; // invisible delimiters to help Excel users
+    var delimLF = "\r\n";
+    var tDelim = "";
+
+    //finds and displays column names
+    for (let i = 0; i < columns; i++) {
+      tDelim = i === columns - 1 ? "" : delim;
+      gridHeadBuffer +=
+        '<td style="width: 100px;">' + values.format[i].name + tDelim + "</td>";
+      columnOrder.push(values.format[i].id);
+    }
+
+    //populates table
+    for (var i = 0; i < rows; i++) {
```

Comment:
This should be a "let i".  Code cleanup since you're in here already.

Reply:
@Pelentan For all of these I'd rather scope code cleanup in a different story. This PR is a literal copy paste from LEAF_Request_Portal/js to /app/libs/js/LEAF, and meant to be a preventative measure to realign files and avoid someone adding new things on the wrong file. I noticed that some work was being done on the unused files, which would definitely cause problems later.


Ideally this would be a symbolic link (until the consolidation "single source of truth" is complete) instead of copy/pasting files, but I didn't look into how this would impact Windows/Linux envs.